aten/src/ATen/native/Convolution.cpp

```diff
@@ -119,6 +121,23 @@ auto ConvParams::view1d_as_2d() -> void {
   }
 }
 
+bool ConvParams::use_cpu_depthwise3x3_winograd(
+    const at::Tensor& input, const at::Tensor& weight) const
+{
+  // Currently only 3x3 depthwise convolutions on tensors of float are supported.
+  return !transposed &&
+         !is_dilated() &&
+         (input.device().type() == c10::DeviceType::CPU) &&
+         (input.scalar_type() == at::kFloat) &&
+         (input.ndimension() == 4) &&
+         (input.size(1) == groups) &&
+         (weight.device().type() == c10::DeviceType::CPU) &&
+         (weight.scalar_type() == at::kFloat) &&
+         (weight.size(0) % input.size(1) == 0) &&
+         (weight.size(2) == 3) &&
```

Comment:
nit: pus size checks in front as they would be the one commonly triggered (after ndimension of course)